lib/elm-test.js

```diff
@@ -444,3 +327,202 @@ ${testFileGlobs.join('\n')}
+      console.error(
+        `\`${command}\` must be run in the same directory as an existing elm.json file! To make one: elm init`
+      );
+      process.exit(1);
+    }
+  };
+
+  program
+    .storeOptionsAsProperties(false)
+    .name('elm-test')
+    .usage('[options] [globs...]')
+    .description(examples)
+    // For example `--seed` and `--fuzz` only make sense for the “tests” command
+    // and could be specified for that command only, but then they won’t show up
+    // in `--help`.
+    .option(
+      '--compiler <path>',
+      'Use a custom path to an Elm executable (default: elm)',
+      undefined
+    )
+    .option(
+      '--seed <int>',
+      'Run with a previous fuzzer seed',
+      parsePositiveInteger('--seed <int>'),
+      Math.floor(Math.random() * 407199254740991) + 1000
+    )
+    .option(
+      '--fuzz <int>',
+      'Run with each fuzz test performing this many iterations',
+      parsePositiveInteger('--fuzz <int>'),
+      100
+    )
+    .option(
+      '--report <json|junit|console>',
+      'Print results to stdout in the given format',
+      parseReport('--report <json|junit|console>'),
+      'console'
+    )
+    .option('--watch', 'Run tests on file changes', false)
+    .version(packageInfo.version, '--version', 'Print version and exit')
+    .helpOption('-h, --help', 'Show help')
+    .addHelpCommand('help [command]', 'Show help');
+
+  program
+    .command('init')
+    .description('Create example tests')
+    .action((cmd) => {
+      if (cmd.args.length > 0) {
+        console.error(
+          `error: init takes no arguments, but got ${
+            cmd.args.length
+          }: ${cmd.args.join(' ')}`
+        );
+        throw process.exit(1);
+      }
+      requireElmJsonFile('init');
+      const options = program.opts();
+      const pathToElmBinary = getPathToElmBinary(options.compiler);
+      switch (pathToElmBinary.tag) {
+        case 'Ok':
+          Install.install(pathToElmBinary.value, 'elm-explorations/test');
+          fs.mkdirpSync('tests');
+          fs.copySync(
+            path.join(__dirname, '..', 'templates', 'tests', 'Example.elm'),
+            'tests/Example.elm'
+          );
+          console.log(
+            '\nCheck out the documentation for getting started at https://package.elm-lang.org/packages/elm-explorations/test/latest'
+          );
+          throw process.exit(0);
+        case 'Error':
+          console.error(pathToElmBinary.error);
+          throw process.exit(1);
+      }
+    });
+
+  program
+    .command('install <package>')
+    .description(
+      'Like `elm install package`, except it installs to "test-dependencies" in your elm.json'
+    )
+    .action(([packageName], cmd) => {
+      if (cmd.args.length > 1) {
+        console.error(
+          `error: install takes one single argument, but got ${
+            cmd.args.length
+          }: ${cmd.args.join(' ')}`
+        );
+        throw process.exit(1);
```

Comment:
Unfortunately one has to check for excessive arguments manually. https://github.com/tj/commander.js/issues/1268